docs/guide/features.md

```diff
@@ -617,6 +617,37 @@ const worker = new Worker(new URL('./worker.js', import.meta.url), {
 })
 ```
 
+:::warning
+
+Make sure to declare the `new URL()` constructor directly inside the `new Worker()` declaration, as done in the above example;
+declaring the URL in a separate variable will cause Vite to ignore the constructor, and the Worker may end up not being included in the built bundle.
+
+Additionaly, all options parameters must be static values (i.e. string literals). Using variables or constants will trigger a build error.
+
+The following example won't work:
+
+```javascript
+const WORKER_NAME = 'MyWorkerName'
+const workerUrl = new URL('./worker.js', import.meta.url)
+// the declaration of the worker will be ignored by Vite due to the URL being declared in a separate variable
+const worker = new Worker(workerUrl, {
+  type: 'module',
+  // the non-static name option will trigger an error
+  name: WORKER_NAME,
+})
+```
+
+Here is how it should be:
+
+```javascript
+const worker = new Worker(new URL('./worker.js', import.meta.url), {
+  type: 'module',
+  name: 'MyWorkerName',
+})
+```
+
+:::
+
```

Comment:
I think we should go with a shorter version, as this warning will be bigger than the docs for the feature if not. It is enough to add a short paragraph:

```suggestion
The worker detection will only work if the `new URL()` constructor is used directly inside the `new Worker()` declaration. Additionally, all options parameters must be static values (i.e. string literals).

```